recipes/bazel/all/conanfile.py

```diff
@@ -0,0 +1,58 @@
+from conans import ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+import shutil
+
+
+class BazelConan(ConanFile):
+    name = "bazel"
+    description = "Bazel is a fast, scalable, multi-language and extensible build system."
+    license = "Apache-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://bazel.build/"
+    topics = ("test", "build", "automation")
+    settings = "os", "arch"
+    no_copy_source = True
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _program_suffix(self):
+        return ".exe" if self.settings.os == "Windows" else ""
+
+    def _chmod_plus_x(self, name):
+        os.chmod(name, os.stat(name).st_mode | 0o111)
+
+    @property
+    def _bazel_filename(self):
+        platform = "darwin" if self.settings.os == "Macos" else str(self.settings.os).lower()
+        return "bazel-{}-{}-{}{}".format(self.version, platform, self.settings.arch, self._program_suffix)
+
+    def source(self):
+        tools.mkdir(self._source_subfolder)
+        for source in self.conan_data["sources"][self.version]:
+            url = source["url"]
+            filename = os.path.join(self._source_subfolder, url[url.rfind("/") + 1:])
+            tools.download(url, filename)
```

Comment:
Seems odd to download two unrelated platforms 🤔

Reply:
we may try to download in build, like in case of MSYS.
still not sure, if we want to install prebuilt binaries

Reply:
Okay, I'll move to build()

I would prefer building from sources too, but we can improve it in the future.